pkg/core/blockchain.go

```diff
@@ -1554,7 +1554,7 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
 		return fmt.Errorf("%w: %d vs %d", ErrTxInvalidWitnessNum, len(t.Signers), len(t.Scripts))
 	}
 	interopCtx := bc.newInteropContext(trigger.Verification, bc.dao, block, t)
-	gasLimit := t.NetworkFee
+	gasLimit := t.NetworkFee - int64(t.Size())*bc.FeePerByte()
```

Comment:
We need a test for it. Calculate the fee for tx and then subtract one satoshi from it, sign and verify.